pkg/nfd-master/node-updater-pool.go

```diff
@@ -71,7 +72,13 @@ func (u *nodeUpdaterPool) processNodeUpdateRequest(queue workqueue.RateLimitingI
 }
 
 func (u *nodeUpdaterPool) runNodeUpdater(queue workqueue.RateLimitingInterface) {
-	for u.processNodeUpdateRequest(queue) {
+	// For tests, re-use the api client from nfd-master
+	cli := u.nfdMaster.k8sClient
+	if u.nfdMaster.kubeconfig != nil {
```

Comment:
I am thinking, should we do the if first and defer to an `else` in case `u.nfdMaster.kubeconfig == nil`

Reply:
This was a short version to initialize the `cli` variable. An alternative would be like:

```golang
	var cli k8sclient.Interface
	if u.nfdMaster.kubeconfig != nil {
		// For normal execution, initialize a separate api client for each updater
		cli = k8sclient.NewForConfigOrDie(u.nfdMaster.kubeconfig)
	} else {
		// For tests, re-use the api client from nfd-master
		cli := u.nfdMaster.k8sClient
	}
```

I'm ok with either

Reply:
I like it, not a blocker

Reply:
I changed the code to what I suggested